stubs/pyOpenSSL/OpenSSL/crypto.pyi

```diff
@@ -180,24 +180,24 @@ class NetscapeSPKI:
     def b64_encode(self) -> bytes: ...
     def get_pubkey(self) -> PKey: ...
     def set_pubkey(self, pkey: PKey) -> None: ...
-    def sign(self, pkey: PKey, digest: bytes) -> None: ...
+    def sign(self, pkey: PKey, digest: str) -> None: ...
     def verify(self, key: PKey) -> bool: ...
 
 def get_elliptic_curves() -> set[_EllipticCurve]: ...
-def get_elliptic_curve(name: Text) -> _EllipticCurve: ...
+def get_elliptic_curve(name: str) -> _EllipticCurve: ...
 def dump_certificate(type: int, cert: X509) -> bytes: ...
 def load_certificate(type: int, buffer: bytes) -> X509: ...
 def dump_certificate_request(type: int, req: X509Req) -> bytes: ...
 def load_certificate_request(type: int, buffer: bytes) -> X509Req: ...
 def dump_privatekey(
-    type: int, pkey: PKey, cipher: bytes | None = ..., passphrase: bytes | Callable[[], bytes] | None = ...
+    type: int, pkey: PKey, cipher: str | None = ..., passphrase: bytes | Callable[[], bytes] | None = ...
 ) -> bytes: ...
-def load_privatekey(type: int, buffer: Text | bytes, passphrase: bytes | Callable[[], bytes] | None = ...) -> PKey: ...
+def load_privatekey(type: int, buffer: str | bytes, passphrase: bytes | Callable[[], bytes] | None = ...) -> PKey: ...
 def dump_publickey(type: int, pkey: PKey) -> bytes: ...
-def load_publickey(type: int, buffer: Text | bytes) -> PKey: ...
+def load_publickey(type: int, buffer: str | bytes) -> PKey: ...
 def dump_crl(type: int, crl: CRL) -> bytes: ...
-def load_crl(type: int, buffer: Text | bytes) -> CRL: ...
-def load_pkcs7_data(type: int, buffer: Text | bytes) -> PKCS7: ...
-def load_pkcs12(buffer: Text | bytes, passphrase: bytes | None = ...) -> PKCS12: ...
-def sign(pkey: PKey, data: Text | bytes, digest: Text | bytes) -> bytes: ...
-def verify(cert: X509, signature: bytes, data: Text | bytes, digest: Text | bytes) -> None: ...
+def load_crl(type: int, buffer: str | bytes) -> CRL: ...
+def load_pkcs7_data(type: int, buffer: str | bytes) -> PKCS7: ...
+def load_pkcs12(buffer: str | bytes, passphrase: bytes | None = ...) -> PKCS12: ...
+def sign(pkey: PKey, data: str | bytes, digest: str) -> bytes: ...
```

Comment:
Idea for future PR: Some arguments, such as `data` here, get passed to `text_to_bytes_and_warn`, which emits a `DeprecationWarning` if you pass a string. We should probably just change them to `bytes` in typeshed, so that the warning happens when type checking rather than at runtime.

Reply:
Im curious, is this a standard thing done in the typing community?

This seems to force application maintainers into one of the following options if they use a deprecated type

- They could add a `# type: ignore` which stops typing of more stuff than we want
- They could not update to the newest version of these stubs and miss out on changes which fixed wrong typings
- They could use the new type, which may or may not be possible depending on the project

Don’t seem very good options to me

Especially because DeprecationWarnings can be catched by unittests.

But i don’t have much experience with typing so maybe this is standard procedure in the community and its not a big deal.

Reply:
We currently don't have a good way to mark deprecated things. There's some discussion about this here: https://github.com/python/typing/issues/1043

For now, we usually keep deprecated things in the spirit of avoiding false positives. This was just a random idea.